shell/platform/linux/fl_standard_message_codec.cc

```diff
@@ -0,0 +1,557 @@
+}
+
+static void write_uint32(GByteArray* buffer, uint32_t value) {
+  g_byte_array_append(buffer, reinterpret_cast<uint8_t*>(&value),
+                      sizeof(uint32_t));
+}
+
+static void write_int32(GByteArray* buffer, int32_t value) {
+  g_byte_array_append(buffer, reinterpret_cast<uint8_t*>(&value),
+                      sizeof(int32_t));
+}
+
+static void write_int64(GByteArray* buffer, int64_t value) {
+  g_byte_array_append(buffer, reinterpret_cast<uint8_t*>(&value),
+                      sizeof(int64_t));
+}
+
+static void write_float64(GByteArray* buffer, double value) {
+  g_byte_array_append(buffer, reinterpret_cast<uint8_t*>(&value),
+                      sizeof(double));
+}
+
+// Checks there is enough data in @buffer to be read.
+static gboolean check_size(GBytes* buffer,
+                           size_t offset,
+                           size_t required,
+                           GError** error) {
+  if (offset + required > g_bytes_get_size(buffer)) {
+    g_set_error(error, FL_MESSAGE_CODEC_ERROR,
+                FL_MESSAGE_CODEC_ERROR_OUT_OF_DATA, "Unexpected end of data");
+    return FALSE;
+  }
+  return TRUE;
+}
+
+// Gets a pointer to the given offset in @buffer.
+static const uint8_t* get_data(GBytes* buffer, size_t* offset) {
+  return static_cast<const uint8_t*>(g_bytes_get_data(buffer, nullptr)) +
+         *offset;
+}
+
+// Read an unsigned 8 bit number from @buffer and writes it to @value.
+// Returns TRUE if successful, otherwise sets an error.
+static gboolean read_uint8(GBytes* buffer,
+                           size_t* offset,
+                           uint8_t* value,
+                           GError** error) {
+  if (!check_size(buffer, *offset, sizeof(uint8_t), error))
+    return FALSE;
+
+  *value = get_data(buffer, offset)[0];
+  (*offset)++;
+  return TRUE;
+}
+
+// Read an unsigned 16 bit integer from @buffer and writes it to @value.
+// Returns TRUE if successful, otherwise sets an error.
+static gboolean read_uint16(GBytes* buffer,
+                            size_t* offset,
+                            uint16_t* value,
+                            GError** error) {
+  if (!check_size(buffer, *offset, sizeof(uint16_t), error))
+    return FALSE;
+
+  *value = reinterpret_cast<const uint16_t*>(get_data(buffer, offset))[0];
+  *offset += sizeof(uint16_t);
+  return TRUE;
+}
+
+// Read an unsigned 32 bit integer from @buffer and writes it to @value.
+// Returns TRUE if successful, otherwise sets an error.
+static gboolean read_uint32(GBytes* buffer,
+                            size_t* offset,
+                            uint32_t* value,
+                            GError** error) {
+  if (!check_size(buffer, *offset, sizeof(uint32_t), error))
+    return FALSE;
+
+  *value = reinterpret_cast<const uint32_t*>(get_data(buffer, offset))[0];
+  *offset += sizeof(uint32_t);
+  return TRUE;
+}
+
+// Reads a signed 32 bit integer from @buffer and writes it to @value.
```

Comment:
Copypasta here and all the later comments; it doesn't write to @value (which doesn't exist)